typescript/cli/src/utils/files.ts

```diff
@@ -0,0 +1,147 @@
+import fs from 'fs';
```

Comment:
is this mostly borrowed from infra? why dupe this?

Reply:
Good question! I don't want to put it in `utils` because it's node-specific and won't work in the browser. We'd have to complicate the `utils` build and export settings to expose different versions for different environments. Feasible but fragile. 

Alternatively we could have a `node-utils` package to DRY this up, but didn't think it'd be worth it